src/services/task-processor.js

```diff
@@ -71,10 +71,10 @@ export default class TaskProcessor {
       }
 
       logger.debug('Saving task');
-      this.taskStore.save(nextTask);
+      await this.taskStore.save(nextTask);
 
     } finally {
-      this.taskLockStore.releaseLock(lock);
+      await this.taskLockStore.releaseLock(lock);
```

Comment:
Very important await :)

Reply:
Oh there were even 2 of them that we had missed 😅